cmake/TestSetup.cmake

```diff
@@ -51,10 +51,17 @@ function(setup_test dir arg extra_labels)
   if (Kokkos_ENABLE_OPENMP)
     set(PARTHENON_KOKKOS_TEST_ARGS "${PARTHENON_KOKKOS_TEST_ARGS} --kokkos-threads=${NUM_OMP_THREADS_PER_RANK}")
   endif()
-  add_test( NAME regression_test:${dir} COMMAND ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" 
-    ${arg} --test_dir "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}"
-    --output_dir "${PROJECT_BINARY_DIR}/tst/regression/outputs/${dir}"
-    --kokkos_args=${PARTHENON_KOKKOS_TEST_ARGS})
+  if (SERIAL_WITH_MPIEXEC)
+    set(PREFIX_SERIAL_REGRESSION_TEST ${TEST_MPIEXEC})
+    set(SUFFIX_SERIAL_REGRESSION_TEST --mpirun ${TEST_MPIEXEC})
+  endif()
+  add_test(
+    NAME regression_test:${dir}
+    COMMAND ${PREFIX_SERIAL_REGRESSION_TEST} ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" 
+      ${arg} --test_dir "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}"
+      --output_dir "${PROJECT_BINARY_DIR}/tst/regression/outputs/${dir}"
+      --kokkos_args=${PARTHENON_KOKKOS_TEST_ARGS}
+      ${SUFFIX_SERIAL_REGRESSION_TEST})
```

Comment:
I think if we are going to use mpi in this function we should probably change the names for the functions, because we already have setup_test_mpi, this can now use mpi but it's not in the name, we could change the function names to setup_test_serial and setup_test_parallel.

Reply:
`setup_serial_test`?

I probably should change `if(SERIAL_WITH_MPI)` to `if(SERIAL_WITH_MPI AND NOT DISABLE_MPI)` or something.